### PR TITLE
Ark scale integration

### PR DESCRIPTION
**Pull request by achimcc**
Use [ark-scale](https://github.com/w3f/ark-scale) by @burdges for (de-)serialization.

### CODE REVIEW

File: curves/bls12_377/src/curves/mod.rs [hunk 1]
```diff
@@ -47,37 +51,44 @@ impl<H: HostFunctions> Bls12Config for Config<H> {
...
     ) -> MillerLoopOutput<Bls12<Self>> {
-        let a: Vec<Vec<u8>> = a
+        let a: ArkScale<Vec<<Bls12<Self> as Pairing>::G1Prepared>> = a
             .into_iter()
             .map(|elem| {
                 let elem: <Bls12<Self> as Pairing>::G1Prepared = elem.into();
-                serialize_argument(elem)
+                elem
             })
-            .collect();
```
Comment (burdges) [inline curves/bls12_377/src/curves/mod.rs:56]: At the moment this does not compile right?

It should read like this I think:
```
let a = a.into_iter().map(ark_ec::pairing::prepare_g1);
let a: Vec<u8> = ark_scale::iter_ark_to_ark_bytes(a).unwrap();
```

File: curves/bls12_381/src/curves/g1.rs [hunk 2]
```diff
@@ -144,36 +145,33 @@ impl<H: HostFunctions> SWCurveConfig for Config<H> {
...
-            .collect();
+        let bases: ArkScale<&[Affine<Self>]> = bases.into();
+        let scalars: ArkScale<&[<Self as CurveConfig>::ScalarField]> = scalars.into();
 
-        let result = H::bls12_381_msm_g1(bases, scalars);
+        let result = H::bls12_381_msm_g1(bases.encode(), scalars.encode()).unwrap();
 
-        let result = deserialize_result::<Affine<Self>>(&result);
-        Ok(result.into())
+        let result = <ArkScale<Projective<Self>> as Decode>::decode(&mut result.clone().as_slice());
```
Comment (burdges) [inline curves/bls12_381/src/curves/g1.rs]: `ark_scale::hazmat::ArkScaleProjective<Projective<Self>>::`

File: curves/bls12_377/src/curves/mod.rs (hunk 1 above)
Comment (achimcc) [reply curves/bls12_377/src/curves/mod.rs:56]: How about:
```rust
let a: ArkScale<Vec<<Bls12<Self> as Pairing>::G1Prepared>> = a
            .into_iter()
            .map(ark_ec::pairing::G1Prepared::from)
            .collect::<Vec<_>>()
            .into();
```

File: curves/bls12_381/src/curves/g1.rs (hunk 2 above)
Comment (achimcc) [reply curves/bls12_381/src/curves/g1.rs]: This was done. Probably forgot to push.

**Pull request merged**